projects/plugins/jetpack/extensions/shared/block-category.native.js

```diff
@@ -0,0 +1,28 @@
+import { getCategories, setCategories, registerBlockCollection } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+import { JetpackLogo } from './icons';
+
+registerBlockCollection( 'jetpack', {
+	title: 'Jetpack powered',
+	icon: <JetpackLogo />,
+} );
```

Comment:
There are only two differences between this native file and the sibling web file: 

1. The [`isWpCom` conditional](https://github.com/Automattic/jetpack/blob/52dc9943f51d013cd1405e0e8f4d66c1326f729b/projects/plugins/jetpack/extensions/shared/block-category.js#L9) is removed (this check relies upon `window` and does not work for the native editor). 
2. Changing the collection title from "Jetpack" to "Jetpack powered." 

A separate native file feels a bit excessive for such small differences, but it is likely for the best so that (1) the web bundle size does not increase with native code and (2) a misleading `isWpCom` no-op is not run for the native editor.

Reply:
@dcalhoun I'm wondering if this should only be enabled for the WordPress app, probably showing the Jetpack logo in the Jetpack app would be a bit redundant, wdyt?

Reply:
+1 to this seeming a bit redundant for the Jetpack app, but not necessarily a blocker to merging if limiting it to only the WP app proves to be complex (especially as the eventual plan is to remove it).

Reply:
> +1 to this seeming a bit redundant for the Jetpack app, but not necessarily a blocker to merging if limiting it to only the WP app proves to be complex (especially as the eventual plan is to remove it).

Good point, I also think that only enabling it for the WP app shouldn't be a blocker.

Reply:
> Are there alternative approaches I am overlooking?

I noticed that in the iOS native side we're using an object named [`AppConfiguration`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) that contains different flags, including `isJetpack` to determine if the app is the Jetpack one. As an example, this flag is being used for determining when the Jetpack banner should be displayed ([reference](https://github.com/wordpress-mobile/WordPress-iOS/blob/ddb0ba2c7e48e6e3989bc8c948592f71da7b4344/WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift#L606-L608)), so we could use that value for this too.

In relation to how we could pass this information to RN side, I was thinking to have a new capability or initial prop to identify the client. This way we could enable/disable the block collection depending on the client (i.e. the app), wdyt?

Reply:
> I noticed that in the iOS native side we're using an object named [`AppConfiguration`](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) that contains different flags, including `isJetpack` to determine if the app is the Jetpack one. 

Agreed. This does seem the most appropriate flag to use. Thanks for sharing. 

> In relation to how we could pass this information to RN side, I was thinking to have a new capability or initial prop to identify the client. This way we could enable/disable the block collection depending on the client (i.e. the app), wdyt?

As I consider it further, setting the app type within `capabilities` feels a bit inappropriate. I imagine it makes more sense to have a `isJetpack` boolean within the editor props/settings. 

I can begin work to limit when the "Jetpack powered" badge displays, but it looks like the consensus is that work should not block merging this work. 👍🏻

Reply:
> As I consider it further, setting the app type within `capabilities` feels a bit inappropriate. I imagine it makes more sense to have a `isJetpack` boolean within the editor props/settings.

True, although this would imply adding it to Gutenberg and hence, add a reference to Jetpack 😅. An alternative could be to generate a different RN bundle for each WordPress and Jetpack app. This way we could set global constants similar to how it's done on the native side with [AppConfiguration](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) for each app. In fact, this option might be necessary in the future for overriding the colors of the editor for example.

> I can begin work to limit when the "Jetpack powered" badge displays, but it looks like the consensus is that work should not block merging this work. 👍🏻

Yep, let's wrap up this task first and then focus on how we can identify in the editor the app that is being run 👍 .

Reply:
> True, although this would imply adding it to Gutenberg and hence, add a reference to Jetpack 😅. An alternative could be to generate a different RN bundle for each WordPress and Jetpack app. This way we could set global constants similar to how it's done on the native side with [AppConfiguration](https://github.com/wordpress-mobile/WordPress-iOS/blob/trunk/WordPress/Jetpack/AppConfiguration.swift) for each app. In fact, this option might be necessary in the future for overriding the colors of the editor for example.

I wondered this myself. I hoped we might be able to strip out the value within `gutenberg-mobile`'s editor setup code, but we may still run into issue where the new property may have to be defined within `react-native-bridge` protocols. 

One alternative might be to define a generic host app identifier, e.g. `hostAppNamespace: 'jetpack'|'wordpress'|'awesome-future-app'`.

Reply:
> Another alternative might be to define a generic host app identifier, e.g. `hostAppNamespace: 'jetpack'|'wordpress'|'awesome-future-app'`.

Yeah, this is aligned to what I [commented](https://github.com/Automattic/jetpack/pull/25092#discussion_r922205803) about passing the editor's client, although I agree that `hostAppNamespace` sounds more accurate.